File: android/src/test/java/apitests/java/CommonApiTests.java
package apitests.java;

import android.app.Activity;
import android.content.Context;

import androidx.annotation.NonNull;

import com.revenuecat.purchases.DangerousSettings;
import com.revenuecat.purchases.Store;
import com.revenuecat.purchases.common.PlatformInfo;
import com.revenuecat.purchases.hybridcommon.CommonKt;
import com.revenuecat.purchases.hybridcommon.ErrorContainer;
import com.revenuecat.purchases.hybridcommon.OnResult;
import com.revenuecat.purchases.hybridcommon.OnResultAny;
import com.revenuecat.purchases.hybridcommon.OnResultList;
import com.revenuecat.purchases.models.GoogleProrationMode;

import java.util.List;
import java.util.Map;

import kotlin.Unit;
import kotlin.jvm.functions.Function1;

@SuppressWarnings({"unused", "deprecation"})
class CommonApiTests {
    private void checkCheckSetAllowSharingAppStoreAccount(boolean enabled) {
        CommonKt.setAllowSharingAppStoreAccount(enabled);
    }

    private void checkGetOfferings(OnResult onResult) {
        CommonKt.getOfferings(onResult);
    }

    private void checkGetProductInfo(List<String> productIDs,
                                     String type,
                                     OnResultList onResult) {
        CommonKt.getProductInfo(productIDs, type, onResult);
    }

    private void checkPurchaseProduct(Activity activity,
                                      String productIdentifier,
                                      String type,
                                      String googleBasePlanId,
                                      String googleOldProductId,
                                      Integer googleProrationMode,
                                      Boolean googleIsPersonalizedPrice,
                                      OnResult onResult) {
        CommonKt.purchaseProduct(
                activity,
                productIdentifier,
                type,
                googleBasePlanId,
                googleOldProductId,
                googleProrationMode,
                googleIsPersonalizedPrice,
                onResult
        );
    }

    private void checkPurchasePackage(Activity activity,
                                      String packageIdentifier,
                                      String offeringIdentifier,
                                      String googleOldProductId,
                                      Integer googleProrationMode,
                                      Boolean googleIsPersonalizedPrice,
                                      OnResult onResult) {
        CommonKt.purchasePackage(
                activity,
                packageIdentifier,
                offeringIdentifier,
                googleOldProductId,
                googleProrationMode,
                googleIsPersonalizedPrice,
                onResult
        );
    }

    private void checkPurchaseSubscriptionOption(Activity activity,
                                                 String productIdentifier,
                                                 String optionIdentifier,
                                                 String googleOldProductId,
                                                 Integer googleProrationMode,
                                                 Boolean googleIsPersonalizedPrice,
                                                 OnResult onResult) {
        CommonKt.purchaseSubscriptionOption(
                activity,
                productIdentifier,
                optionIdentifier,
                googleOldProductId,
                googleProrationMode,
                googleIsPersonalizedPrice,
                onResult
        );
    }

    private void checkGetAppUserId() {
        String appUserId = CommonKt.getAppUserID();
    }

    private void checkRestorePurchases(OnResult onResult) {
        CommonKt.restorePurchases(onResult);
    }

    private void checkLogIn(String appUserId, OnResult onResult) {
        CommonKt.logIn(appUserId, onResult);
    }

    private void checkLogOut(OnResult onResult) {
        CommonKt.logOut(onResult);
    }

    private void checkSetDebugLogsEnabled(boolean enabled) {
        CommonKt.setDebugLogsEnabled(enabled);
    }

    private void checkSetLogLevel(String level) {
        CommonKt.setLogLevel(level);
    }

    private void checkSetLogHandler(Function1<? super Map<String, String>, Unit> callback) {
        CommonKt.setLogHandler(callback);
    }

    private void checkSetLogHandlerWithOnResult() {
        CommonKt.setLogHandlerWithOnResult(new OnResult() {
            @Override
            public void onReceived(@NonNull Map<String, ?> map) {
            }

            @Override
            public void onError(@NonNull ErrorContainer errorContainer) {
            }
        });
    }

    private void checkSetProxyURLString(String proxyUrlString) {
        CommonKt.setProxyURLString(proxyUrlString);
    }

    private void checkGetProxyURLString() {
        String proxyUrlString = CommonKt.getProxyURLString();
    }

    private void checkGetCustomerInfo(OnResult onResult) {
        CommonKt.getCustomerInfo(onResult);
    }

    private void checkSyncPurchases() {
        CommonKt.syncPurchases();
    }

    private void checkIsAnonymous() {
        boolean isAnonymous = CommonKt.isAnonymous();
    }

    private void checkSetFinishTransactions(boolean enabled) {
        CommonKt.setFinishTransactions(enabled);
    }

    private void checkCheckTrialOrIntroductoryPriceEligibility(List<String> productIdentifiers) {
        Map<String, Map<String, Object>> result = CommonKt.checkTrialOrIntroductoryPriceEligibility(productIdentifiers);
    }

    private void checkInvalidateCustomerInfoCache() {
        CommonKt.invalidateCustomerInfoCache();
    }

    private void checkCanMakePayments(Context context, List<Integer> features, OnResultAny<Boolean> onResult) {
        CommonKt.canMakePayments(context, features, onResult);
    }

    private void checkConfigure(Context context,
                                String apiKey,
                                String appUserId,
                                Boolean observerMode,
                                PlatformInfo platformInfo,
                                Store store,
                                DangerousSettings dangerousSettings) {
        CommonKt.configure(context, apiKey, appUserId, observerMode, platformInfo);
        CommonKt.configure(context, apiKey, appUserId, observerMode, platformInfo, store);
        CommonKt.configure(context, apiKey, appUserId, observerMode, platformInfo, store, dangerousSettings);
    }

    private void checkGetPromotionalOffer() {
        ErrorContainer errorContainer = CommonKt.getPromotionalOffer();
    }

    private void checkErrorContainer(Integer code, String message, Map<String, Object> info) {
        ErrorContainer errorContainer = new ErrorContainer(code, message, info);
        Integer storedCode = errorContainer.getCode();
        String storedMessage = errorContainer.getMessage();
        Map<String, Object> storedAny = errorContainer.getInfo();
    }
}

File: android/src/test/java/apitests/kotlin/com/revenuecat/purchases/hybridcommon/CommonApiTests.kt
package apitests.kotlin.com.revenuecat.purchases.hybridcommon

import android.app.Activity
import android.content.Context
import com.revenuecat.purchases.DangerousSettings
import com.revenuecat.purchases.Store
import com.revenuecat.purchases.common.PlatformInfo
import com.revenuecat.purchases.hybridcommon.ErrorContainer
import com.revenuecat.purchases.hybridcommon.OnResult
import com.revenuecat.purchases.hybridcommon.OnResultAny
import com.revenuecat.purchases.hybridcommon.OnResultList
import com.revenuecat.purchases.hybridcommon.canMakePayments
import com.revenuecat.purchases.hybridcommon.checkTrialOrIntroductoryPriceEligibility
import com.revenuecat.purchases.hybridcommon.configure
import com.revenuecat.purchases.hybridcommon.getAppUserID
import com.revenuecat.purchases.hybridcommon.getCustomerInfo
import com.revenuecat.purchases.hybridcommon.getOfferings
import com.revenuecat.purchases.hybridcommon.getProductInfo
import com.revenuecat.purchases.hybridcommon.getPromotionalOffer
import com.revenuecat.purchases.hybridcommon.getProxyURLString
import com.revenuecat.purchases.hybridcommon.invalidateCustomerInfoCache
import com.revenuecat.purchases.hybridcommon.isAnonymous
import com.revenuecat.purchases.hybridcommon.logIn
import com.revenuecat.purchases.hybridcommon.logOut
import com.revenuecat.purchases.hybridcommon.purchasePackage
import com.revenuecat.purchases.hybridcommon.purchaseProduct
import com.revenuecat.purchases.hybridcommon.purchaseSubscriptionOption
import com.revenuecat.purchases.hybridcommon.restorePurchases
import com.revenuecat.purchases.hybridcommon.setAllowSharingAppStoreAccount
import com.revenuecat.purchases.hybridcommon.setDebugLogsEnabled
import com.revenuecat.purchases.hybridcommon.setFinishTransactions
import com.revenuecat.purchases.hybridcommon.setLogHandler
import com.revenuecat.purchases.hybridcommon.setLogHandlerWithOnResult
import com.revenuecat.purchases.hybridcommon.setLogLevel
import com.revenuecat.purchases.hybridcommon.setProxyURLString
import com.revenuecat.purchases.hybridcommon.showInAppMessagesIfNeeded
import com.revenuecat.purchases.hybridcommon.syncPurchases
import com.revenuecat.purchases.models.InAppMessageType

@Suppress("unused", "DEPRECATION", "LongParameterList", "UNUSED_VARIABLE")
private class CommonApiTests {
    fun checkSetAllowSharingAppStoreAccount(allow: Boolean) {
        setAllowSharingAppStoreAccount(allow)
    }

    fun checkGetOfferings(onResult: OnResult) {
        getOfferings(onResult)
    }

    fun checkGetProductInfo(
        productIDs: List<String>,
        type: String,
        onResult: OnResultList,
    ) {
        getProductInfo(productIDs, type, onResult)
    }

    fun checkPurchaseProduct(
        activity: Activity?,
        productIdentifier: String,
        type: String,
        googleBasePlanId: String?,
        googleOldProductId: String?,
        googleProrationMode: Int?,
        googleIsPersonalizedPrice: Boolean?,
        presentedOfferingIdentifier: String?,
        onResult: OnResult,
    ) {
        purchaseProduct(
            activity,
            productIdentifier,
            type,
            googleBasePlanId,
            googleOldProductId,
            googleProrationMode,
            googleIsPersonalizedPrice,
            presentedOfferingIdentifier,
            onResult,
        )
    }

    fun checkPurchasePackage(
        activity: Activity?,
        packageIdentifier: String,
        offeringIdentifier: String,
        googleOldProductId: String?,
        googleProrationMode: Int?,
        googleIsPersonalizedPrice: Boolean?,
        onResult: OnResult,
    ) {
        purchasePackage(
            activity,
            packageIdentifier,
            offeringIdentifier,
            googleOldProductId,
            googleProrationMode,
            googleIsPersonalizedPrice,
            onResult,
        )
    }

    fun checkPurchaseSubscriptionOption(
        activity: Activity?,
        productIdentifier: String,
        optionIdentifier: String,
        googleOldProductId: String?,
        googleProrationMode: Int?,
        googleIsPersonalizedPrice: Boolean?,
        presentedOfferingIdentifier: String?,
        onResult: OnResult,
    ) {
        purchaseSubscriptionOption(
            activity,
            productIdentifier,
            optionIdentifier,
            googleOldProductId,
            googleProrationMode,
            googleIsPersonalizedPrice,
            presentedOfferingIdentifier,
            onResult,
        )
    }

    fun checkGetAppUserId() {
        val appUserId: String = getAppUserID()
    }

    fun checkRestorePurchases(onResult: OnResult) {
        restorePurchases(onResult)
    }

    fun checkLogIn(
        appUserID: String,
        onResult: OnResult,
    ) {
        logIn(appUserID, onResult)
    }

    fun checkLogOut(onResult: OnResult) {
        logOut(onResult)
    }

    fun checkSetDebugLogsEnabled(enabled: Boolean) {
        setDebugLogsEnabled(enabled)
    }

    fun checkSetLogLevel(level: String) {
        setLogLevel(level)
    }

    fun checkSetLogHandler(callback: (logDetails: Map<String, String>) -> Unit) {
        setLogHandler(callback)
    }

    fun checkSetLogHandlerWithOnResult() {
        setLogHandlerWithOnResult(object : OnResult {
            override fun onReceived(map: Map<String, *>) {}
            override fun onError(errorContainer: ErrorContainer) {}
        })
    }

    fun checkSetProxyURLString(proxyURLString: String?) {
        setProxyURLString(proxyURLString)
    }

    fun checkGetProxyURLString() {
        val proxyURLString: String? = getProxyURLString()
    }

    fun checkGetCustomerInfo(onResult: OnResult) {
        getCustomerInfo(onResult)
    }

    fun checkSyncPurchases() {
        syncPurchases()
    }

    fun checkIsAnonymous() {
        val isAnonymous: Boolean = isAnonymous()
    }

    fun checkSetFinishTransactions(enabled: Boolean) {
        setFinishTransactions(enabled)
    }

    fun checkCheckTrialOrIntroductoryPriceEligibility(productIdentifiers: List<String>) {
        val result: Map<String, Map<String, Any>> = checkTrialOrIntroductoryPriceEligibility(
            productIdentifiers,
        )
    }

    fun checkInvalidateCustomerInfoCache() {
        invalidateCustomerInfoCache()
    }

    fun checkCanMakePayments(
        context: Context,
        features: List<Int>,
        onResult: OnResultAny<Boolean>,
    ) {
        canMakePayments(context, features, onResult)
    }

    fun checkShowInAppMessagesIfNeeded(activity: Activity?) {
        showInAppMessagesIfNeeded(activity)
        showInAppMessagesIfNeeded(activity, listOf(InAppMessageType.BILLING_ISSUES))
        showInAppMessagesIfNeeded(activity, null)
    }

    fun checkConfigure(
        context: Context,
        apiKey: String,
        appUserID: String?,
        observerMode: Boolean?,
        platformInfo: PlatformInfo,
        store: Store,
        dangerousSettings: DangerousSettings,
        shouldShowInAppMessagesAutomatically: Boolean?,
    ) {
        configure(context, apiKey, appUserID, observerMode, platformInfo)
        configure(context, apiKey, appUserID, observerMode, platformInfo, store, dangerousSettings)
        configure(
            context,
            apiKey,
            appUserID,
            observerMode,
            platformInfo,
            store,
            dangerousSettings,
            shouldShowInAppMessagesAutomatically,
        )
    }

    fun checkGetPromotionalOffer() {
        val errorContainer: ErrorContainer = getPromotionalOffer()
    }

    fun checkErrorContainer(code: Int, message: String, info: Map<String, Any?>) {
        val errorContainer = ErrorContainer(code, message, info)
        val storedCode = errorContainer.code
        val storedMessage = errorContainer.message
        val storedInfo = errorContainer.info
    }
}
